### DISCUSSION TITLE
Automatically clear unused images

### DISCUSSION THREAD

**Discussion by martpie** [avitorio/outstatic#268] [Ideas]
I just spent 10 minutes clearing unused images in `public/` on a repo using Outstatic. It would be awesome if removed images from content + cover image, they were also removed from the `public/` folder.

I realize that is a complex-ish feature to build, especially since there can be a significant perf hit at scale. At the same time, Outstatic is really meant for small-medium websites. Instead of doing it at every save, there could be a button alongside "rebuild metadata" to scan all content for used images, and remove the unused ones?

This goes against the feature of a "media gallery", but I'm not sure this is something you'll ever want to build.

**Comment by avitorio** [comment]
Hey @martpie,

The image gallery feature is coming to v1.5. The issue I see with cleaning images automatically is how to make sure images that are used outside of mdx files are still kept.

Thoughts?

**Reply by martpie** [reply]
Gallery makes this a bit harder, and even maybe less desirable.

My really dumb assumption would be that outstatic can query the `public/` folder from the git repo and establish a list of images. Then, as Oustatic is aware (is it?) of the different collections and field type (images custom fields, mdx, cover image etc), it can scan through all content and look for images mardkown tags, or references, establish a list of those, the delete the unused one from git.

Bonus point for letting users preview the deleted images and cancel things if needed.

As a rather costly operation, this would probably make more sense to have this as a kind of manual operation users can do every once in a while, rather than done automatically.

All of this of course to avoid having a growth of the repo size and also for the sake of sanity.

Maybe it does not make sense, I really don't know, I'll let you judge :)

**Reply by avitorio** [reply]
I like the idea of an automated clean up, but at the same time I don't risk deleting files the user doesn't want deleted. 
It's a tricky feature, as images the user wants to keep may not be referenced on the .mdx files at all, like the site logo, etc...

I think something that would make this a lot easier would be adding the ability to multi-select images for deletion on the Media Gallery. Which is something I want to add in the near future. Perhaps that would be a more meaningful feature right now.
